c/main.c

```diff
@@ -0,0 +1,150 @@
+#include <deltachat.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif
+
+void handle_message(dc_context_t *context, int chat_id, int message_id) {
+  dc_chat_t *chat = dc_get_chat(context, chat_id);
+
+  if (dc_chat_get_type(chat) == DC_CHAT_TYPE_SINGLE) {
+    dc_msg_t *msg = dc_get_msg(context, message_id);
+
+    if (!msg) {
+      return;
+    }
+
+    dc_send_text_msg(context, chat_id, dc_msg_get_text(msg));
+    dc_msg_unref(msg);
+  }
+
+  dc_chat_unref(chat);
+}
+
+void *event_handler(void *context) {
+  dc_event_emitter_t *emitter = dc_get_event_emitter(context);
+  dc_event_t *event;
+  while ((event = dc_get_next_event(emitter)) != NULL) {
+    // use the event as needed, e.g. dc_event_get_id() returns the type.
+    // once you're done, unref the event to avoid memory leakage:
+    int event_type = dc_event_get_id(event);
+
+    if (event_type == DC_EVENT_ERROR || event_type == DC_EVENT_INFO ||
+        event_type == DC_EVENT_WARNING) {
+      char *message = dc_event_get_data2_str(event);
+
+      switch (event_type) {
+      case DC_EVENT_ERROR:
+        printf("[Error] %s\n", message);
+        break;
+      case DC_EVENT_INFO:
+        printf("[Info] %s\n", message);
+        break;
+      case DC_EVENT_WARNING:
+        printf("[Warn] %s\n", message);
+        break;
+      }
+    } else if (event_type == DC_EVENT_CONFIGURE_PROGRESS) {
+
+      int progress = dc_event_get_data1_int(event);
+      char *comment = dc_event_get_data2_str(event);
```

Comment:
same here, the result of dc_event_get_data2_str() must be passed to dc_str_unref() after usage.